src/Tizen.Sensor/Tizen.Sensor/Sensor.cs

```diff
@@ -545,5 +604,600 @@ private void DestroyHandles()
+            };
+
+            int error = Interop.SensorProvider.ProviderSetStartCb(_sensorProviderHandle, _sensorProviderStartCallBack, userData);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in setting start callback of sensor provider");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetStartCb Failed");
+            }
+
+        }
+
+        private void ProviderSetStopCb(IntPtr userData)
+        {
+            _sensorProviderStopCallBack = (IntPtr _provider, Int64 _userData) =>
+            {
+                ProviderStopEventArgs e = new ProviderStopEventArgs(_userData)
+                {
+                };
+                ProviderStoped?.Invoke(null, e);
+            };
+
+            int error = Interop.SensorProvider.ProviderSetStopCb(_sensorProviderHandle, _sensorProviderStopCallBack, userData);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in setting stop callback of sensor provider");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetStopCb Failed");
+            }
+
+        }
+
+        private void ProviderSetIntervalChangedCb(IntPtr userData)
+        {
+            _sensorProviderIntervalChangedCallBack = (IntPtr _provider, uint IntervalMs, Int64 _userData) =>
+            {
+                ProviderIntervalChangedEventArgs e = new ProviderIntervalChangedEventArgs(_userData,IntervalMs)
+                {
+                };
+                ProviderIntervalChanged?.Invoke(null, e);
+            };
+
+            int error = Interop.SensorProvider.ProviderSetIntervalChangedCb(_sensorProviderHandle, _sensorProviderIntervalChangedCallBack, userData);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in setting interval callback of sensor provider");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetIntervalChangedCb Failed");
+            }
+
+        }
+
+        private void ProviderPublish(Interop.SensorProvider.SensorEventS sensorEventS)
+        {
+            int error = Interop.SensorProvider.ProviderPublish(_sensorProviderHandle, sensorEventS);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in sensor provider publish");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderPublish Failed");
+            }
+
+        }
+
+        /// <summary>
+        /// Adds a callback function to be invoked when a new sensor is added.
+        /// </summary>
+        /// <param name="userData">
+        /// The user data to be passed to the callback function.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void AddSensorAddedCB(IntPtr userData)
```

Comment:
USE event, DO NOT USE callback

Reply:
API name is changed to SetSensorAddedCB. This is API for setting callback .